### DISCUSSION TITLE
Prisma groupby question

### DISCUSSION THREAD

**Discussion by srikumarkp** [prisma/prisma#21563] [Q&A]
### Question

Have a Prisma client which was using raw SQL previously and migrating to use groupBy - 
previous code had an aggregation as shown below:
```sql
SUM(avg_session_duration * sessions) / SUM(sessions) as avgSessionDuration,
SUM(new_sessions_percent * sessions) / SUM(sessions) as percentNewSessions
```
Is there a way to migrate this to use `groupBy` feature in Prisma API ?

Thanks in Advance,
Sri

### Information about Prisma Schema, Client Queries and Environment (optional)

// Add any relevant Prisma Client queries here
```
prisma client version - 5.4.1
```

**Comment by ludralph** [comment]
Hi @srikumarkp 👋 

Thank you for raising this question.

The SQL query you provided performs calculations across all rows without any grouping. It calculates the weighted average of `avg_session_duration` and `new_sessions_percent` based on the `sessions` column. Are you sure this fits the criteria for using Prisma's `groupBy` feature?

**Reply by srikumarkp** [reply]
hi @ludralph,

Thanks for the response.

Sorry, the Sql query was incomplete:

Is of the form below:

**select x,y,z
SUM(avg_session_duration * sessions) / SUM(sessions) as avgSessionDuration,
SUM(new_sessions_percent * sessions) / SUM(sessions) as percentNewSessions
from table where = whereFiltere group by x,yz, order by x;**

Is there a way i can model this query using Prisma groupby feature ?

**Reply by ludralph** [reply]
Hi @srikumarkp 

This SQL cannot be expressed using Prisma's `groupBy`. You will need to use [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) to perform the query. You can also leverage [SafeQL](https://safeql.dev/compatibility/prisma.html) to write raw queries in a type-safe way.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
